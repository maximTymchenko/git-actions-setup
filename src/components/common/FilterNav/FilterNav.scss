@import '../../../styles/vars';

.filter-nav {
    height: 100%;
    max-height: 430px;
    box-shadow: 9px 9px 1px 1px rgba(57, 156, 203, 0.4);
    border-bottom: 2px solid $secondary;
    top: 123px;
    padding-top: 30px;
    background: $background__secondary;
    width: 100%;
    position: fixed;
    left: 0;
    z-index: 99;

    &__wrapper {
        width: 100%;
        max-width: $container__size;
        margin: 0 auto;
    }
}
