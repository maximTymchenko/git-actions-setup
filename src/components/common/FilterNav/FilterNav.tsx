import React, { FC, useState } from 'react';
import PrimaryNav from '../PrimaryNav';

const FilterNav: FC = () => {
    interface INavLinks {
        activeLink: null | any;
        links: [
            {
                id: number;
                title: string;
                page?: string | undefined;
            },
        ];
    }

    interface INavArr {
        activeLink?: null;
        links: {
            id: number;
            array: INavLinks;
        }[];
    }

    interface ILanguage {
        activeLink: null;
        links: {
            id: number;
            title: string;
        }[];
    }

    const language: ILanguage = {
        activeLink: null,
        links: [
            { id: 1, title: '' },
            { id: 2, title: '' },
        ],
    };

    const [navLinks, setNavLinks] = useState<INavLinks>({
        activeLink: null,
        links: [
            { id: 1, title: 'Services' },
            { id: 2, title: 'Industries' },
            { id: 3, title: 'Expertise' },
            { id: 4, title: 'Engagement' },
        ],
    });

    const menuLinksServices: INavLinks = {
        activeLink: null,
        links: [
            { id: 1, title: 'Web development', page: '/what-we-do/web-development/' },
            { id: 2, title: 'Cloud services', page: '/what-we-do/cloud-services/' },
            { id: 3, title: 'Team extension', page: '/what-we-do/team-extension/' },
            { id: 4, title: 'AR & VR', page: '/what-we-do/ar-&-vr/' },
            { id: 5, title: 'Retail & E-commerce', page: '/what-we-do/retaile-and-e-commerce/' },
            { id: 6, title: 'Mobile development', page: '/what-we-do/mobile-development/' },
        ],
    };

    const menuLinksIndustries: INavLinks = {
        activeLink: null,
        links: [
            { id: 1, title: 'Retail & E-commerce' },
            { id: 2, title: 'Construction' },
            { id: 3, title: 'Finance' },
            { id: 4, title: 'Media' },
        ],
    };

    const menuLinksExpertise: INavLinks = {
        activeLink: null,
        links: [
            { id: 1, title: 'SaaS product' },
            { id: 2, title: 'Chatbots' },
            { id: 3, title: 'AR & VR' },
            { id: 4, title: 'Blockchain' },
            { id: 5, title: 'Digital enterprise ' },
            { id: 6, title: 'Drone' },
            { id: 7, title: 'R&D' },
        ],
    };

    const menuLinksEngagement: INavLinks = {
        activeLink: null,
        links: [
            { id: 1, title: 'On-demand team extension' },
            { id: 2, title: 'Fully managed IT services' },
        ],
    };

    const [navLinksArray, setNavLinksArray] = useState<INavArr>({
        activeLink: null,
        links: [
            { id: 0, array: menuLinksServices },
            { id: 1, array: menuLinksIndustries },
            { id: 2, array: menuLinksExpertise },
            { id: 3, array: menuLinksEngagement },
        ],
    });
    return (
        <div data-testid="filter-list">
            <PrimaryNav
                navLinks={navLinks}
                navLinksArray={navLinksArray}
                language={language}
                type="filter"
            />
        </div>
    );
};
export default FilterNav;
