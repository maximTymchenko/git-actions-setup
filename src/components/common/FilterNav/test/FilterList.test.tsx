import React from 'react'
import FilterNav from '../FilterNav'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'

it('renders elem correctly', () => {
  const { getByTestId } = render(<FilterNav />)

  expect(getByTestId('filter-list')).toBeTruthy()
})

it('renders without unexpected changes', () => {
  const tree = renderer.create(<FilterNav />).toJSON()
  expect(tree).toMatchSnapshot()
})
