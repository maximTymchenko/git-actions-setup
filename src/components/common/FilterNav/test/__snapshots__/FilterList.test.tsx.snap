// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div
  className="filter-nav"
  data-testid="filter-list"
>
  <div
    className="filter-nav__wrapper"
  >
    <div
      className="test-nav__wrapper"
      data-testid="primary-nav"
    >
      <span
        className="test-nav__title"
        onClick={[Function]}
      >
        Services
      </span>
    </div>
    <div
      className="test-nav__wrapper"
      data-testid="primary-nav"
    >
      <span
        className="test-nav__title"
        onClick={[Function]}
      >
        Industries
      </span>
    </div>
    <div
      className="test-nav__wrapper"
      data-testid="primary-nav"
    >
      <span
        className="test-nav__title"
        onClick={[Function]}
      >
        Expertise
      </span>
    </div>
    <div
      className="test-nav__wrapper"
      data-testid="primary-nav"
    >
      <span
        className="test-nav__title"
        onClick={[Function]}
      >
        Engagement
      </span>
    </div>
  </div>
</div>
`;
